### PR TITLE
HorizontalView: children.map to React.Children.map

### PR DESCRIPTION
**Pull request by kialysa**
Before: required to pass in an array of components. if last component is false, renders nothing

After: array is not required. if any component is false, it is filtered out

Checklist:

- [ ] Test added / Snapshots updated
- [ ] Documentation added / updated

### CODE REVIEW

File: src/behaviour/HorizontalView.jsx [hunk 1]
```diff
@@ -65,22 +65,24 @@ class HorizontalView extends React.Component {
     }
   }
 
-  componentWillReceiveProps(nextProps) {
-    const oldChildren = this.props.children
-    const nextChildren = nextProps.children
+  static getDerivedStateFromProps(
+    { children: nextProps },
+    { children: previousState }
+  ) {
```
Comment (kialysa) [inline src/behaviour/HorizontalView.jsx]: is nextProps.children or previousState.children ever undefined?
is this better?
```
  { children: nextProps = [] },
 { children: previousState = [] }
```

**Comment by mosch** [comment]
Sweet! I think it would make sense to add a unit test for this. That would help preventing that this pops up again.

File: src/behaviour/HorizontalView.jsx (hunk 1 above)
Comment (mosch) [reply src/behaviour/HorizontalView.jsx]: Not sure actually. I think the only case where it could be undefined is if you do <HorizontalView /> like this.

**Comment by kialysa** [comment]
@mosch  you got it 👍

File: src/behaviour/HorizontalView.jsx (hunk 1 above)
Comment (kialysa) [reply src/behaviour/HorizontalView.jsx]: Ok, then I'll leave it.

**Comment by kialysa** [comment]
@mosch i've written the tests now! let me know if there's sth im missing

**Pull request merged**